### PR TITLE
ft:buyer track order status(create/get/delete)

### PR DESCRIPTION
**Pull request by emilenfc**
# Description

In this feature, buyer can add create order status ofter he create cart, and he can track order status, Admin can change the order status and send update to the buyer profile

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/database/models/Order.model.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        key: "id",
+      },
+    },
+    items: {
+      type: DataTypes.JSONB,
+      allowNull: false,
+    },
+    status: {
+      type: DataTypes.ENUM("pending", "shipped", "delivered"),
+      defaultValue: "pending",
```
Comment (shyakadavis) [inline src/database/models/Order.model.ts:28]: It's not a breaking change, but can you try capitalizing the enums for consistency elsewhere.

e.g `Pending` , `Shipped`, `Delivered`

File: src/controllers/Order.controller.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import { Request, Response } from "express";
+import Cart from "../database/models/Cart.model";
+import Order from "../database/models/Order.model";
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:3]: I don't know if it's from personal preference, but abstracting the logic of communicating with the db to a separate folder like that of `Services`, would be much better easier to read. 🤷🏽‍♂️ 

e.g. writing a Order.service.ts and putting there any function that directly deals with the db, like creating a order, tracking an order ...

File: src/controllers/Order.controller.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+export const updatedOrderStatus = async (req: Request, res: Response) => {
+  try {
+    const orderId = req.params.orderId;
+    const order = await Order.findOne({
+      where: {
+        id: orderId,
+      },
+    });
+    if (!order) {
+      return res.status(404).json({ message: "Order not found" });
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:131]: These response messages are not the ones we using to be honest. This is way off track, wouldn't you agree❓

File: src/controllers/Order.controller.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    const updatedOrder = await order.update({
+      status: req.body.status,
+    });
+    const updatedOrderSt = {
+      currentStatus: updatedOrder.dataValues.status,
+      expectedDeliveryDate: updatedOrder.dataValues.expectedDeliveryDate,
+    };
+    return res.status(200).json({ message: "status updated", updatedOrderSt });
+  } catch (err) {
+    return res.status(500).json({ message: "Internal server error" });
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:142]: Even the ones in the catch block, please look into how we are writing those.

File: test/order.spec.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
+import request from "supertest";
+import app from "../src/app";
+
+jest.setTimeout(1500000);
+
+describe("🛒 📦 CART UNIT", () => {
```
Comment (shyakadavis) [inline test/order.spec.ts:6]:
```suggestion
describe(" 🚛 ✈️  ORDERS UNIT", () => {
```

**Review by SergeInGithub — COMMENTED** [review]
Try making the changes @shyakadavis suggested please!!

File: src/controllers/Order.controller.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    if (items.length === 0) {
+      res.status(400).json({
+        message: "No product available To make order",
+        status: 400,
+        success: false,
+      });
+    }
+    const order = await createOrderService(userId, items);
+    res.status(201).json({
+      orderData: order,
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:25]:
```suggestion
      data: order,
```

File: src/controllers/Order.controller.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+      data: orders,
+      status: 200,
+      success: true,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      res.status(404).json({
+        status: 404,
+        success: false,
+        message: error.message,
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:56]:
```suggestion
        message: 'Error viewing all orders',
        error: error.message,
```

File: src/controllers/Order.controller.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+      });
+    } else {
+      res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error  while getting order",
+      });
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:63]:
```suggestion
       console.log('Unexpected error when viewing all orders: ', error);
```

File: src/controllers/Order.controller.ts [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+// GET request to retrieve order status information
+export const getOrderStatus = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
+    const orderId = req.params.orderId;
+
+    const orderStatus = await getOrderStatusService(userId, orderId);
+    res.status(200).json({
+      orderStatus,
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:98]:
```suggestion
      data: orderStatus,
```

File: src/controllers/Order.controller.ts [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+      return res.status(404).json({
+        status: 404,
+        success: false,
+        message: err.message,
+      });
+    } else {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error retrieving order status",
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:113]:
```suggestion
        message: "Error retrieving order status",
        error: err.message,
```

File: src/controllers/Order.controller.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+      orderStatus,
+      status: 200,
+      success: true,
+    });
+  } catch (err) {
+    if (err instanceof Error) {
+      return res.status(404).json({
+        status: 404,
+        success: false,
+        message: err.message,
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:107]:
```suggestion
        message: 'Error getting all orders status',
        error: error.message
```

File: src/controllers/Order.controller.ts [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+      });
+    } else {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error retrieving order status",
+      });
```
Comment (shyakadavis) [inline src/controllers/Order.controller.ts:114]:
```suggestion
      console.log('Unexpected error when viewing all orders status', error);
```

**Review by shyakadavis — DISMISSED** [review]
LGTM

**Review by musllim — DISMISSED** [review]
this looks good to me

**Pull request closed without merging**